aten/src/ATen/native/BatchLinearAlgebra.cpp

```diff
@@ -921,8 +921,7 @@ static Tensor& linalg_solve_out_info(Tensor& result, Tensor& infos, const Tensor
   // - 2-dimensional (2D) tensor or batch of 2D tensors (matrix case)
   // original torch.solve supported only the matrix case, while NumPy works for both cases
   // for the batched input we need to be able to distinguish them
-  auto expected_batched_rhs_shape = IntArrayRef(input.sizes().data(), input.dim()-1);  // input.shape[:-1]
-  bool vector_case = other.dim() == 1 || (input.dim()-1 == other.dim() && other.sizes().equals(expected_batched_rhs_shape));
+  bool vector_case = linalg_solve_is_vector_rhs(input, other);
 
   bool is_batched_column_major = false;
   if (vector_case) {
```

Comment:
nit: `is_batched_column_major` does not have a comment, and the variable is used a bit far away from its definition making the code a touch harder to understand.

Reply:
Changes in this function, `linalg_solve_out_info`, were introduced only to move the code for determining vector/matrix case to a separate common function to be used both in `linalg_solve` and `linalg_lstsq`. I don't want to introduce any further changes to this function as it's not related to the scope of this PR.